_rules/aria-required-context-role-ff89c9.md

```diff
@@ -0,0 +1,284 @@
+- Some user agents and assistive technologies ignore empty elements, which means they are not presented to all users. However, since this is handled inconsitently across user agents and assistive technologies, this rule is applicable to empty elements. 
+
+## Background
+
+- [Understanding Success Criterion 1.3.1: Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
+- [Required Context Role](https://www.w3.org/TR/wai-aria-1.1/#scope)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+Element with role `listitem` is contained within its required context role `list`, expressed as an explicit role.
+
+```html
+<div role="list">
+    <div role="listitem">List item 1</div>
+</div>
+```
+
+#### Passed Example 2
+
+Element with role `listitem` is contained within its required context role `list`, through the implicit role of `ul`.
+
+```html
+<ul>
+    <div role="listitem">List item 1</div>
+</ul>
+```
+
+#### Passed Example 3
+
+Element contained within its required context role even though it is not a direct child of the context role.
+
+```html
+<div role="list">
+    <div>
+         <div>
+             <div>
+                <div role="listitem">List item 1</div>
+             </div>
+         </div>
+    </div>
+</div>
+```
+
+#### Passed Example 4
+
+`aria-owns` used to give the target element the right context role.
+
+```html
+<div role="list" aria-owns="id1"></div>
+<div id="id1" role="listitem">List item 1</div>
+```
+
+#### Passed Example 5
+
+`aria-owns` trumps ownership by closest ancestor, giving the element with role of `listitem` the correct context role.
+
+```html
+<div role="list" aria-owns="id1">
+  <div role="tabpanel">
+    <div id="id1" role="listitem">List item 1</div>
+   </div>
+</div>
+```
+
+#### Passed Example 6
+
+Since implicit ownership can cross shadow boundaries, the element with the semantic role of `listitem` is contained within its required context role `list`.
+
+```html
+<div id="host" role="list">
+</div>
+
+<script>
+  const host = document.querySelector("#host");
+  const root = host.attachShadow({ mode: "open" });
+
+  root.innerHTML = `
+    <div role="listitem">List item 1</div>
+  `;
+</script>
+```
+#### Passed Example 7
+
+Element with role `listitem` is contained within its required context role `list`, expressed as an explicit role. Even though the element with role `listitem` is empty, it is still applicable to this rule.
+
+```html
+<div role="list">
+    <div role="listitem"></div>
+</div>
+```
+
+### Failed
+
+#### Failed Example 1
+
+No context role.
+
+```html
+<div role="listitem">List item 1</div>
+```
+
+#### Failed Example 2
+
+Wrong context role.
+
+```html
+<div role="tablist">
+    <div role="listitem">List item 1</div>
+</div>
+```
+
+#### Failed Example 3
+
+Element not contained within its required context role.
+
+```html
+<div role="list"></div>
+    <div role="listitem">List item 1</div>
```

Comment:
This shouldn't indent. Can you please run prettier on this rule before you merge it?

Reply:
Fixed in this and one other test case